### PR TITLE
[meson] Re-add build_by_default to embedded targets.

### LINKED ISSUE #1052: [meson] Some build artifacts missing in meson builds
That's how build-bin looks in [this](https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=5925&view=logs&j=924ff16e-cbb4-544c-ec55-782cc11682e2&t=5f421ada-e94c-5c26-2e38-f7572553c9a8) pipeline run:

```
|-- build-bin
|   |-- hw
|   |   `-- top_earlgrey
|   |       `-- Vtop_earlgrey_verilator
|       |   `-- sim-verilator
|       |       |-- boot_rom
|       |       |   |-- boot_rom.elf
|       |       |   `-- chip_info.h
|       |       |-- examples
|       |       |   |-- hello_usbdev
|       |       |   |   `-- hello_usbdev.elf
|       |       |   `-- hello_world
|       |       |       `-- hello_world.elf
|       |       |-- exts
|       |       |   `-- common
|       |       |       `-- libriscv_crt.a
|       |       |-- lib
|       |       |   |-- libflash_ctrl_ot.a
|       |       |   |-- libgpio_ot.a
|       |       |   |-- libhmac_ot.a
|       |       |   |-- libirq_ot.a
|       |       |   |-- liblog_ot.a
|       |       |   |-- libmem_ot.a
|       |       |   |-- libpinmux_ot.a
|       |       |   |-- librv_timer_ot.a
|       |       |   |-- libspid_ot.a
|       |       |   |-- libuart_ot.a
|       |       |   `-- libusb_ot.a
|       |       `-- tests
|       |           |-- flash_ctrl
|       |           |   `-- flash_test.elf
|       |           |-- hmac
|       |           |   `-- sha256_test.elf
|       |           `-- rv_timer
|       |               `-- rv_timer_test.elf
|       `-- host
|           |-- spiflash
|           |   `-- spiflash
|           `-- vendor
|               `-- libmpsse.a
```

The tree looks strange as vmem and bin files are missing, but some "useless" binaries are there (like libmpsse.a). The "too many files" is a cosmetic problem, the "vmem and bin files missing" is critical.

### PR DESCRIPTION
**Pull request by mcy**
Apparently, build_by_default's default value is dependent on both the target type (custom_target vs executable) and Meson version. Even worse, Meson's documentation merely states that it affects "running plain ninja", and *not* that it also removes the target from the "all" target.

Fixes #1052.

### CODE REVIEW

**Comment by imphil** [comment]
CI fails because the sha256 test fails.

```
DEBUG:functional_verilator_test:fd#9: b'Commit ID:  274306ecef72c623f59e8d9334986013798e4131\r\n'
DEBUG:functional_verilator_test:fd#9: b'Build Date: 2019-11-22, 15:58:33\r\n'
DEBUG:functional_verilator_test:fd#9: b'INFO: Boot ROM initialisation has completed, jump into flash!\r\n'
DEBUG:functional_verilator_test:fd#9: b'Running SHA256 test\r\n'
DEBUG:functional_verilator_test:fd#9: b'FAIL!\r\n'
```


I'd be OK with disabling this test for now and filing a follow-up.

**Comment by mcy** [comment]
Done. I filed #1058 to track the breakage.

File: ci/run_verilator_pytest.sh [hunk 1]
```diff
@@ -14,16 +14,19 @@ SW_BUILD_PATH="${SW_BUILD_PATH:-$SW_BUILD_DEFAULT}"
 
 BOOT_ROM_TARGET="boot_rom/boot_rom.vmem"
 
-TEST_TARGETS=("tests/flash_ctrl/flash_test.vmem"
-  "tests/hmac/sha256_test.vmem"
-  "tests/rv_timer/rv_timer_test.vmem"
+TEST_TARGETS=(
+  "tests/flash_ctrl/flash_test.vmem"
+  # XXX: Temporarially disabled, since these tests are failing at HEAD.
```
Comment (gkelly) [inline ci/run_verilator_pytest.sh]: nit: `Temporarily`.

Comment (imphil) [reply ci/run_verilator_pytest.sh]: Please also add the issue number to the comment.
XXX -> TODO

**Pull request merged**